### PR TITLE
[@pollyjs] Add missing adapter and persister packages

### PR DESCRIPTION
**Pull request by offirgolan**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/pollyjs__adapter-puppeteer/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Type definitions for @pollyjs/adapter-puppeteer 4.0
+// Project: https://github.com/netflix/pollyjs/tree/master/packages/@pollyjs/adapter-puppeteer
+// Definitions by: offirgolan <https://github.com/offirgolan>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+import Adapter from '@pollyjs/adapter';
+
+export default class PuppeteerAdapter extends Adapter {}
```
Comment (weswigham) [inline types/pollyjs__adapter-puppeteer/index.d.ts]:
```suggestion
class PuppeteerAdapter extends Adapter {}
export = PuppeteerAdapter;
```

If you look at their [runtime behavior](https://runkit.com/embed/597ogce4iypk) they transpile a `default` into a `module.exports` assignment, so the types need to reflect the runtime reality, not the original source.

File: types/pollyjs__persister-local-storage/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Type definitions for @pollyjs/persister-local-storage 4.0
+// Project: https://github.com/netflix/pollyjs/tree/master/packages/@pollyjs/persister-local-storage
+// Definitions by: offirgolan <https://github.com/offirgolan>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+import Persister from '@pollyjs/persister';
+
+export default class LocalStoragePersister extends Persister {}
```
Comment (weswigham) [inline types/pollyjs__persister-local-storage/index.d.ts]: Same here; [at runtime](https://runkit.com/embed/sqgcmm893xy7) the module doesn't use a default, but rather assigns to the module as a whole.

**Review by weswigham — CHANGES_REQUESTED** [review]
Use `export =` in the export assignments and then add `esModuleInterop: true` in the `tsconfig`s for the tests.

File: types/pollyjs__adapter-puppeteer/index.d.ts (hunk 1 above)
Comment (offirgolan) [reply types/pollyjs__adapter-puppeteer/index.d.ts]: Updated. There are other polly packages that need this change. Should I address that in a follow up PR or include it here?

**Review by weswigham — APPROVED** [review]

**Pull request merged**